crates/storage/src/codec.rs

```diff
@@ -16,19 +16,22 @@ pub mod primitive;
 pub mod raw;
 
 /// The trait is usually implemented by the encoder that stores serialized objects.
-pub trait Encoder {
+pub trait Encoder<'a> {
     /// Returns the serialized object as a slice.
     fn as_bytes(&self) -> Cow<[u8]>;
+
+    /// Consumes the encoder, returning the serialized object as a slice.
+    fn into_bytes(self) -> Cow<'a, [u8]>;
 }
 
 /// The trait encodes the type to the bytes and passes it to the `Encoder`,
 /// which stores it and provides a reference to it. That allows gives more
 /// flexibility and more performant encoding, allowing the use of slices and arrays
 /// instead of vectors in some cases. Since the [`Encoder`] returns `Cow<[u8]>`,
 /// it is always possible to take ownership of the serialized value.
-pub trait Encode<T: ?Sized> {
+pub trait Encode<T: ?Sized>: 'static {
```

Comment:
I'm not exactly sure why we need the `'static` bound here.

Reply:
Right no this is just a leftover from a previous attempt to fix the lifetime issues with the `Encode` trait, but in the current solution this is redundant. Thanks for noticing!

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2396/commits/c3b72642dacdbbe44fff3c6a3a95d68beb3ff6c5